src/lammpsio/snapshot.py

```diff
@@ -107,18 +107,64 @@ def from_hoomd_gsd(cls, frame):
             if numpy.any(snap.molecule < 0):
                 warnings.warn("Some molecule IDs are negative, remapping needed.")
 
-        if (
-            frame.bonds.N > 0
-            or frame.angles.N > 0
-            or frame.dihedrals.N > 0
-            or frame.impropers.N > 0
-        ):
-            warnings.warn("Conversion of topology from gsd is not supported")
-        type_map = {typeid + 1: i for typeid, i in enumerate(frame.particles.types)}
-
-        return snap, type_map
-
-    def to_hoomd_gsd(self, type_map=None):
+        # set particle label
+        label_map_particle = None
+        if frame.particles.types is not None:
+            label_map_particle = {
+                typeid + 1: i for typeid, i in enumerate(frame.particles.types)
+            }
+        snap.label = LabelMap(map=label_map_particle)
+
+        if frame.bonds.N != 0:
+            snap.bonds = Bonds(N=frame.bonds.N)
+            snap.bonds.typeid = frame.bonds.typeid + 1
+            snap.bonds.members = frame.bonds.group + 1
+
+            # set bond label
+            label_map_bond = {
+                typeid + 1: i for typeid, i in enumerate(frame.bonds.types)
+            }
+            snap.bonds.label = LabelMap(map=label_map_bond)
+
+        if frame.angles.N != 0:
+            snap.angles = Angles(N=frame.angles.N)
+            snap.angles.typeid = frame.angles.typeid + 1
+            snap.angles.members = frame.angles.group + 1
+
+            # set angle label
+            label_map_angle = {
+                typeid + 1: i for typeid, i in enumerate(frame.angles.types)
+            }
+            snap.angles.label = LabelMap(map=label_map_angle)
+
+        if frame.dihedrals.N != 0:
+            snap.dihedrals = Dihedrals(N=frame.dihedrals.N)
+            snap.dihedrals.typeid = frame.dihedrals.typeid + 1
+            snap.dihedrals.members = frame.dihedrals.group + 1
+
+            # set dihedral label
+            label_map_dihedral = {
+                typeid + 1: i for typeid, i in enumerate(frame.dihedrals.types)
+            }
+            snap.dihedrals.label = LabelMap(map=label_map_dihedral)
+
+        if frame.impropers.N != 0:
+            snap.impropers = Impropers(N=frame.impropers.N)
+            snap.impropers.typeid = frame.impropers.typeid + 1
+            snap.impropers.members = frame.impropers.group + 1
+
+            # set improper label
+            label_map_improper = {
+                typeid + 1: i for typeid, i in enumerate(frame.impropers.types)
+            }
+            snap.impropers.label = LabelMap(map=label_map_improper)
+
+        return snap, snap.label
```

Comment:
This should still be a dictionary of particle types for backwards compatibility
```suggestion
        return snap, label_map_particle
```